app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/NotificationListener.java

```diff
@@ -351,12 +361,109 @@ public void onNotificationPosted(StatusBarNotification sbn) {
+        long start = System.currentTimeMillis();
+
+        DBHandler db;
+
+        try {
+            db = GBApplication.acquireDB();
+        } catch (GBException e) {
+            LOG.error("Could not acquire DB.", e);
+            return true;
+        }
+
+        List<String> wordsList = new ArrayList<>();
+
+        NotificationFilterDao notificationFilterDao = db.getDaoSession().getNotificationFilterDao();
+        NotificationFilterEntryDao notificationFilterEntryDao = db.getDaoSession().getNotificationFilterEntryDao();
+
+        Query<NotificationFilter> query = notificationFilterDao.queryBuilder().where(NotificationFilterDao.Properties.AppIdentifier.eq(packageName.toLowerCase())).build();
+        NotificationFilter notificationFilter = query.unique();
+
+        if (notificationFilter == null) {
+            LOG.debug("No Notification Filter found");
+            return true;
+        }
+
+        LOG.debug("Loaded notification filter for '{}'", packageName);
+        Query<NotificationFilterEntry> queryEntries = notificationFilterEntryDao.queryBuilder().where(NotificationFilterEntryDao.Properties.NotificationFilterId.eq(notificationFilter.getId())).build();
+
+        List<NotificationFilterEntry> filterEntries = queryEntries.list();
+
+        if (BuildConfig.DEBUG) {
+            LOG.info("Database lookup took '{}' ms", System.currentTimeMillis() - start);
+        }
+
+        if (!filterEntries.isEmpty()) {
+            for (NotificationFilterEntry temp : filterEntries) {
+                wordsList.add(temp.getNotificationFilterContent());
+                LOG.debug("Loaded filter word: " + temp.getNotificationFilterContent());
+            }
+        }
+
+        return shouldContinueAfterFilter(body, wordsList, notificationFilter);
+    }
+
+    boolean shouldContinueAfterFilter(@NonNull String body, @NonNull List<String> wordsList, @NonNull NotificationFilter notificationFilter) {
+
+        LOG.debug("Mode: '{}' Submode: '{}' WordsList: '{}'", notificationFilter.getNotificationFilterMode(), notificationFilter.getNotificationFilterSubMode(), wordsList);
+
+        boolean allMode = notificationFilter.getNotificationFilterSubMode() == NOTIFICATION_FILTER_SUBMODE_ALL;
+
+        switch (notificationFilter.getNotificationFilterMode()) {
+            case NOTIFICATION_FILTER_MODE_BLACKLIST:
+                if (allMode) {
+                    for (String word : wordsList) {
+                        if (!body.contains(word)) {
+                            LOG.info("Not every word was found, blacklist has no effect, processing continues.");
+                            return true;
+                        }
+                    }
+                    LOG.info("Every word was found, blacklist has effect, processing stops.");
+                    return false;
+                } else {
+                    boolean notContainsAny = !StringUtils.containsAny(body, wordsList.toArray(new CharSequence[0]));
```

Comment:
Please try to avoid negative identifiers. Logic with built-in negation makes things unnecessarily hard to read/grasp.